tests/unit/backend/layers/business/test_business.py

```diff
@@ -835,6 +835,34 @@ def test_get_dataset_version_from_canonical(self):
             dataset_version = self.business_logic.get_dataset_version_from_canonical(DatasetId(str(uuid.uuid4())))
             self.assertIsNone(dataset_version)
 
+    def test_get_prior_published_versions_for_dataset(self):
+        """
+        Given a canonical dataset id, return all its DatasetVersions that have been part of published CollectionVersions
+        """
+        self.initialize_published_collection()
+        published_version = self.initialize_published_collection()
+        collection_id = published_version.collection_id
+        dataset = published_version.datasets[0]
+        # Revision 1 (to publish)
+        collection_version_id = self.business_logic.create_collection_version(collection_id).version_id
+        new_dataset_version_id, _ = self.business_logic.ingest_dataset(
+            collection_version_id, "http://fake.url", None, dataset.version_id
+        )
+        self.business_logic.publish_collection_version(collection_version_id)
+        # Revision 2 (not to publish)
+        unpublished_collection_version_id = self.business_logic.create_collection_version(collection_id).version_id
+        unpublished_dataset_version_id, _ = self.business_logic.ingest_dataset(
+            unpublished_collection_version_id, "http://fake.url", None, new_dataset_version_id
+        )
+
+        version_history = self.business_logic.get_prior_published_versions_for_dataset(dataset.dataset_id)
+        version_ids = [version.version_id for version in version_history]
+        # Check that only published datasets appear
+        self.assertEqual(len(version_history), 2)
+        self.assertIn(dataset.version_id, version_ids)
+        self.assertIn(new_dataset_version_id, version_ids)
+        self.assertNotIn(unpublished_dataset_version_id, version_ids)
```

Comment:
nit / tell me how you feel: I've started building sets of ids to run a single `assertEqual` on: expected vs received. Fewer comparisons to keep track of since set takes care of number and inclusion/exclusion of relevant ids.

Reply:
Also applies to the test further up